src/network/networkd.c

```diff
@@ -69,17 +70,15 @@ static int run(int argc, char *argv[]) {
         /* Always create the directories people can create inotify watches in.
          * It is necessary to create the following subdirectories after drop_privileges()
          * to support old kernels not supporting AmbientCapabilities=. */
-        r = mkdir_safe_label("/run/systemd/netif/links", 0755, UID_INVALID, GID_INVALID, MKDIR_WARN_MODE);
-        if (r < 0)
-                log_warning_errno(r, "Could not create runtime directory 'links': %m");
-
-        r = mkdir_safe_label("/run/systemd/netif/leases", 0755, UID_INVALID, GID_INVALID, MKDIR_WARN_MODE);
-        if (r < 0)
-                log_warning_errno(r, "Could not create runtime directory 'leases': %m");
-
-        r = mkdir_safe_label("/run/systemd/netif/lldp", 0755, UID_INVALID, GID_INVALID, MKDIR_WARN_MODE);
-        if (r < 0)
-                log_warning_errno(r, "Could not create runtime directory 'lldp': %m");
+        FOREACH_STRING(p,
+                       "/run/systemd/netif/links/",
+                       "/run/systemd/netif/leases/",
+                       "/run/systemd/netif/lldp/",
+                       "/var/lib/systemd/network/dhcp-server-lease/") {
```

Comment:
hmm, so networkd is an early boot service, i.e. might be running super early, when /var/ is not mounted yet.

so i think we probably shouldn't bother with creating the dir here. I mean, we only create this because of inotify, right, but people should probably continue to watch /run/systemd/netif/leases/ for that, no? after all the data in that dir and in "/var/lib/systemd/network/dhcp-server-lease/" is pretty much the same, except the latter is in json, right?

Reply:
oh, hmm, so othe other dir is for the client side, right, and this is for the server side?

so they might actually watch this... hmm, dunno then.

not sure what to do about this...